app/src/main/java/com/omkarmoghe/pokemap/MainActivity.java

```diff
@@ -104,7 +106,15 @@ protected void onStop() {
 
     @Override
     public void onBackPressed() {
-        this.finish();
+        // As request, when the back button is pressed, we should pop the previous value
+        // (using immediate to execute immediate pop)
+        manager.popBackStackImmediate();
+
+        // Grab current amount of fragments in the stack, if 0, means that we can close the app
+        int backStackCount = manager.getBackStackEntryCount();
```

Comment:
@omkarmoghe the code I added pops the stack immediately as the user presses the back button. It will then check whether the new count of the popped stack is 0 and closes the app. So you can see that when you click back (with this commit) the map fragment is popped and it shows a brief grey screen as the app is closed.

However, I am taking account for possible fragments being added into the stack in the near future, where if we had a stack size of 3, pressing back will pop the stack, and essentially close that fragment and committing the changes (popBackStackImmediate()). The app should only close when there are 1 fragments left, and if there's 2, such as when a few users above noted in the PR, it should close the login fragment (if there is one) instead of closing the app, and it should show the map fragment (if it was loaded in the fragment stack)